### PR TITLE
Maven: Change logic to check if a version is released

### PR DESCRIPTION
**Pull request by rebelagentm**
Previously, we parsed an HTML page looking for the name of a dependency to be present.  This was causing failures with Nexus repositories that were returning a non-browsable page.  This PR changes the logic to instead check the headers to see if the dependency exists.

Fixes https://github.com/dependabot/feedback/issues/746

### CODE REVIEW

File: maven/lib/dependabot/maven/update_checker/version_finder.rb [hunk 1]
```diff
@@ -140,18 +140,15 @@ def released?(version)
           @released_check[version] =
             repositories.any? do |repository_details|
               url = repository_details.fetch("url")
-              response = Excon.get(
+              response = Excon.head(
```
Comment (feelepxyz) [inline maven/lib/dependabot/maven/update_checker/version_finder.rb]: Oh looks like this request needs to be stubbed out in a bunch of tests, looks like this might not be that bad, just need to check for places where we already stub out the old request url and change it to include the `jar` file, example in the update checker: https://github.com/dependabot/dependabot-core/blob/master/maven/spec/dependabot/maven/update_checker_spec.rb#L58-L61

Comment (rebelagentm) [reply maven/lib/dependabot/maven/update_checker/version_finder.rb]: I started fixing this, but I'm unsure as to why the example on line 86 is still failing.  I changed the`stub_request` to be `:head` and I added `metadata` to `:dependency_requirements`.

File: maven/spec/dependabot/maven/update_checker_spec.rb [hunk 2]
```diff
@@ -59,15 +65,16 @@
     "https://repo.maven.apache.org/maven2/"\
     "com/google/guava/guava/23.6-jre/"
```
Comment (feelepxyz) [inline maven/spec/dependabot/maven/update_checker_spec.rb]: I think these version URLs need to change to include the file we added, e.g. `guava-23.6-jre.jar`

**Comment by rebelagentm** [comment]
@feelepxyz I fixed almost all of the tests.  There are still two tests in `version_finder_spec.rb` that are failing (lines 95 and 347).  I suspect that it has to do with the fixtures being HTML files, which we are no longer using due to this change.

**Comment by feelepxyz** [comment]
@rebelagentm ah yeah the tests are a bit hard to follow, looks like we are changing the stub for the latest version to have an html response without a jar file. It would be good to get rid of all the html fixtures and `let`s, e.g. https://github.com/dependabot/dependabot-core/blob/master/maven/spec/dependabot/maven/update_checker/version_finder_spec.rb#L63-L65

For the failing tests, we should change the stub_request to respond with a 404 for the latest version and 200 for the earlier version: https://github.com/dependabot/dependabot-core/blob/master/maven/spec/dependabot/maven/update_checker/version_finder_spec.rb#L82-L93
Also rather than `gsub`ing the `maven_central_version_files_url` it would be better to redefine the whole url in that test to make it easier to spot what's going on.

Let me know if you need any help on this!
